magda-web-client/src/Components/RequestDataset/RequestFormLogic.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import RequestFormTemplate from "./RequestFormTemplate";
+import Alert from "./Alert";
+import { correspondenceApiUrl, correspondenceApiReportUrl } from "../../config";
+
+export default class RequestFormLogic extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            successResult: false,
+            posted: false
+        };
+    }
+
+    /**
+     * handles the logic of submitting form
+     * @param {Object} data form submitted data from child component "RequestFormTemplate.js"
+     */
+    handleSubmit = data => {
```

Comment:
This whole thing should probably be happening in a Redux action - we'll leave it for now and make another ticket for it.